### DISCUSSION TITLE
[@types/node] v20.11.8 error TS2502: 'crypto' is referenced directly or indirectly in its own type annotation.

### DISCUSSION THREAD

**Discussion by yusinto** [DefinitelyTyped/DefinitelyTyped#68354] [Issues with a @types package] [answered]
Hi there, I'm reporting a bug in `@types/node` version 20.11.8. Using typescript version 5.2.2, I get the following error:

> ../../../node_modules/@types/node/crypto.d.ts(4478,13): error TS2502: 'crypto' is referenced directly or indirectly in its own type annotation.

The code responsible as referenced above looks like below:

<img width="956" alt="Screenshot 2024-01-27 at 10 04 29 PM" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/1593077/84c2e3b7-7af9-40c7-ac5b-778ba6cf9186">

### Workaround

Downgrading to @types/node version 20.11.7 solves the issue because the problematic global crypto declaration block was absent.

```json
  // package.json
  "resolutions": {
    "@types/node": "20.11.7"
  }
```

[a comment by an automated account is not shown]

**Comment by ghost** [comment, marked as answer]
Fixed in PR: #68353
